modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/builder/ObjectTypeSchemaDelegate.java

```diff
@@ -410,6 +419,16 @@ private TopLevelElement registerAbstractElement(QName typeQName, DslElementSynta
     return abstractElement;
   }
 
+  private QName resolveSubstitutionGroup(SubstitutionGroup userConfiguredSubstitutionGroup) {
+    String namespaceUri = builder.getNamespaceUri(userConfiguredSubstitutionGroup.getPrefix());
```

Comment:
is better to put all this logic in a `builder.resolveSubstitutionGroup` and not having the QName being crafted in the delegate.